File: 이진영/프로그래머스/개인정보 수집 유효기간.js
const MIN_YEAR = 2000;
const MAX_MONTH = 12;
const MAX_DAY = 28;
function solution(today, terms, privacies) {
    let answer = []
    const today_n = getNum(today);
    const termObj = {};
    terms.forEach(v => {
        const [name, month] = v.split(" ");
        termObj[name] = +month;
    })
    
    privacies.forEach((pr, idx) => {
        const [dateStr, termName] = pr.split(" ");
        const start_n = getNum(dateStr);
        const end_n = addTerm(start_n, termObj[termName]);
        if (today_n > end_n) {
            answer.push(idx + 1);
        }
    })
    
    return answer;
}

const getNum = (str) => {
    const [year, month, day] = str.split(".").map(Number);
    return ((year - MIN_YEAR) * MAX_MONTH + month) * MAX_DAY + day;
}

const addTerm = (num, term) => {
    return num + term * MAX_DAY - 1;
}

File: 현세미/프로그래머스/개인정보 수집 유효기간.py
#프로그래머스 - 개인정보 수집 유효기간
def solution(today, terms, privacies):
    def date_to_days(date):
        year, month, day = map(int, date.split('.'))
        return year * 12 * 28 + month * 28 + day #년월일을 일수로 계산

    today_days = date_to_days(today)

    term_dict = {}
    for term in terms:
        kind, month = term.split()
        term_dict[kind] = int(month) * 28

    result = []

    for i, privacy in enumerate(privacies):
        date, kind = privacy.split()
        collected_days = date_to_days(date)
        expiry_days = collected_days + term_dict[kind]

        if today_days >= expiry_days:
            result.append(i + 1)
    
    return result
